### ISSUE TITLE
[java] New rule: UnnecessaryBlock

### ISSUE THREAD

**Issue by UncleOwen** [pmd/pmd#6622]
>Also wondering if [UnnecessaryBlock](https://pmd.github.io/pmd/pmd_rules_ecmascript_codestyle.html#unnecessaryblock) should be ported from JS to Java

_Originally posted by @zbynek in https://github.com/pmd/pmd/pull/6523#issuecomment-4172417987_

**Proposed Rule Name:** UnnecessaryBlock

**Proposed Category:** Code Style.

**Description:** A block is unnecessary, if it is

a) not a class body/method body/body of a control statement/initializer block/... AND
b) not used to restrict the scope of a variable.

**Code Sample:** 

```java
public class Foo {
    public boolean bad(Object o) {
        {
            if(obj == null) return false;
        }
        return true;
    }

    public boolean good(Object o) {
        if(obj == null) return false;
        return true;
    }
}
```
